test/unit/media.js

```diff
@@ -250,4 +254,54 @@ test('should handle unsupported sources with the source hanlder API', function()
 
   tech.setSource('');
   ok(usedNative, 'native source handler was used when an unsupported source was set');
-});
\ No newline at end of file
+});
+
+test('should emulate the video element\'s behavior for currentSrc when src is set', function(){
+  var mockPlayer = {
+    off: this.noop,
+    trigger: this.noop
+  };
+  var sourceA = { src: 'foo.mp4', type: 'video/mp4' };
+  var sourceB = { src: '', type: 'video/mp4' };
+
+  // Define a new tech class
+  var Tech = videojs.MediaTechController.extend();
+
+  // Extend Tech with source handlers
+  vjs.MediaTechController.withSourceHandlers(Tech);
+
+  // Create an instance of Tech
+  var tech = new Tech(mockPlayer);
+
+  // Create source handlers
+  var handler = {
+    canHandleSource: function(source){
+      return 'probably';
+    },
+    handleSource: function(s, t){return {};}
+  };
+
+  Tech.registerSourceHandler(handler);
+
+  // Pass a source through the source handler process of a tech instance
+  tech.setSource(sourceA);
+
+  // Test that currentSource_ is not immediately specified
+  strictEqual(tech.currentSource_, undefined, 'sourceA was not stored immediately');
```

Comment:
If it's a breaking change and we feel this is useful, we should consider it quickly so we can add it to 5.0 before it's too late. Though, really, now would be the time to just iterate through the major versions quickly, once it settles down, we should move more carefully.